pkg/basic/eigentrust.go

```diff
@@ -89,8 +111,15 @@ func Compute(
 	ap.ScaleVec(a, p)
 	tm1 := time.Now()
 	durPrep, tm0 := tm1.Sub(tm0), tm1
+	if flatTailStats == nil {
+		flatTailStats = &FlatTailStats{}
+	}
+	flatTailStats.Length = 0
+	flatTailStats.Threshold = 1
+	flatTailStats.DeltaNorm = 1
+	flatTailStats.Ranking = nil
```

Comment:
Should these initializations be happening inside the 'nil' check conditional block ?

Reply:
Ignore my earlier question. It looks like the variables are computed fresh in this function and it does not matter what the caller passes.